rest/client/rest.go

```diff
@@ -237,3 +237,26 @@ func (c *communicatorImpl) GetDistrosList(ctx context.Context) ([]model.APIDistr
 
 	return distros, nil
 }
+
+func (c *communicatorImpl) GetKeysList(ctx context.Context) ([]model.APIPubKey, error) {
+	info := requestInfo{
+		method:  get,
+		version: apiVersion2,
+		path:    "keys",
+	}
+
+	resp, client_err := c.request(ctx, info, "")
```

Comment:
Messing with this request stuff shouldn't be necessary - you can get the user by using Context.Value()

Reply:
I wouldn't expect the client to associate values with the context. (and contexts shouldn't hide part of the external API) 

Having said that, I think this is fine, given above comment.